mypy/main.py

```diff
@@ -528,6 +514,24 @@ def add_invertible_flag(flag: str,
         print("Warning: --no-fast-parser no longer has any effect.  The fast parser "
               "is now mypy's default and only parser.")
 
+    # Infer Python version and/or executable if one is not given
+    if special_opts.python_executable is not None and special_opts.python_version is not None:
+        py_exe_ver = _python_version_from_executable(special_opts.python_executable)
+        if py_exe_ver != special_opts.python_version:
+            print('Error: Python version and executable are mismatched.')
```

Comment:
A more detailed error message would be nice. You should use `parser.error` here too instead of `print`